src/cache.rs

```diff
@@ -1,38 +1,57 @@
 use crate::content_hash::ContentHash;
 use crate::interns::{FileId, Interns};
+use anyhow::{Context, Result};
+use byteorder::LittleEndian;
 use std::collections::HashMap;
 use std::fs::{self, Metadata};
-use std::io;
 use std::path::Path;
+use zerocopy::byteorder::{I64, U32, U64};
+use zerocopy::{AsBytes, FromBytes, LayoutVerified, Unaligned};
 
 /// File metadata key, based on https://apenwarr.ca/log/20181113
 ///
 /// TODO: Define a different structure for this on Windows.
-#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug)]
-#[cfg(target_family = "unix")]
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug, FromBytes, AsBytes, Unaligned)]
+#[repr(C)]
+#[cfg(any(target_family = "unix", target_endian = "little"))]
```

Comment:
I think we want `all` here rather than `any`, right? Otherwise this is saying "if it's UNIX *or* it's little-endian"

(Also, I think in this context, `all` would be implied, so we could just delete `any`.)